torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -289,6 +289,51 @@ control_plane::RegisterHandler dumpHandler{
           "application/octet-stream");
     }};
 
+control_plane::RegisterHandler jsonDumpHandler{
+    "dump_nccl_trace_json",
+    [](const control_plane::Request& req, control_plane::Response& res) {
+      const auto params = req.params();
+      size_t validParamCount = 0;
+
+      // valid params
+      const std::string includeCollectivesStr = "includecollectives";
+      const std::string onlyActiveStr = "onlyactive";
+
+      std::unordered_map<std::string, bool> expectedParams = {
```

Comment:
nit: maybe name 'processedParams' or something so its clear that this is reused for defaults and merged with user inputs?